hack/verify-yamllint.sh

```diff
@@ -23,9 +23,14 @@ readonly KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 
 cd "${KUBE_ROOT}"
 
+python3 -m venv env
+source env/bin/activate
```

Comment:
might need to .gitignore? also might want to either do this repo wide with a requirements.txt or scope an env specifically to this (call it "yamlenv" or something idk?)

Reply:
🤦 I actually had a related change locally that I forgot to push. I'm reusing the same name as we're using in our makefile so this should also be covered by .gitignore now. Not running the full thing because 100% of our requirements.txt is for docs generation and irrelevant for this specific yamllint script.

https://github.com/kubernetes-sigs/gateway-api/blob/335be6a634e733ac64ad47e17a7548600cec393e/Makefile#L160-L170